### PR TITLE
fix(forms): (TypeScript) ensure `data` type is inherited in onSubmit

### PR DESCRIPTION
**Pull request by tujoworker**
Make this possible (again):
Screenshot 2024-05-07 at 11 58 19

This type test fails when not met:
Screenshot 2024-05-07 at 12 11 42

### CODE REVIEW

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Handler/info.mdx [hunk 1]
```diff
@@ -41,6 +43,51 @@ function Component() {
...
+// Method #2
+const submitHandler = (data: MyDataSet) => {
+  console.log(data.lastName)
+}
+function MyForm() {
+  return <Form.Handler data={data} onSubmit={submitHandler} />
+}
+
+// Method #3
+import type { OnSubit } from '@dnb/eufemia/extensions/forms'
```
Comment (andlbrei) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Handler/info.mdx:82]:
```suggestion
import type { OnSubmit } from '@dnb/eufemia/extensions/forms'
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Handler/info.mdx [hunk 2]
```diff
@@ -41,6 +43,51 @@ function Component() {
...
+const submitHandler: OnSubit<MyDataSet> = (data) => {
```
Comment (andlbrei) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Handler/info.mdx:83]:
```suggestion
const submitHandler: OnSubmit<MyDataSet> = (data) => {
```

**Review by joakbjerk — APPROVED** [review]
Works as intended when testing locally 👍

**Pull request merged**